File: frontend/src/types/product.ts
export interface Product {
  id: string;
  product_name: string | null;
  carrier_name: string | null;
  carrier_rating: string | null;
  product_type: string | null;
  state_availability: string[] | null;
  surrender_period: number | null;
  base_rate: number | null;
  cap_rate: number | null;
  participation_rate: number | null;
  annual_fee: number | null;
  has_bonus: boolean | null;
  bonus_percentage: number | null;
  bonus_vesting_years: number | null;
  minimum_investment: number | null;
  maximum_investment: number | null;
  issue_ages_min: number | null;
  issue_ages_max: number | null;
  created_at: string | null;
  updated_at: string | null;
}

export interface ProductListResponse {
  items: Product[];
  total: number;
  page: number;
  limit: number;
  total_pages: number;
}

export interface IncomeRider {
  id: string;
  rider_name: string | null;
  rider_fee: number | null;
  deferral_bonus_rate: number | null;
  payout_percentage_single_age_65: number | null;
  payout_percentage_single_age_70: number | null;
  payout_percentage_single_age_75: number | null;
  payout_percentage_joint_age_65: number | null;
  payout_percentage_joint_age_70: number | null;
  payout_percentage_joint_age_75: number | null;
  lifetime_guarantee: boolean | null;
  inflation_protection: boolean | null;
}

export interface IndexOption {
  id: string;
  index_name: string | null;
  index_type: string | null;
  cap_rate: number | null;
  participation_rate: number | null;
  spread_fee: number | null;
  floor_rate: number | null;
}

export interface HistoricalPerformance {
  id: string;
  year: number | null;
  credited_rate: number | null;
  index_return: number | null;
  effective_return: number | null;
}

File: backend/app/schemas/product.py
from pydantic import BaseModel
from uuid import UUID
from typing import Optional, List
from datetime import datetime


class IncomeRiderResponse(BaseModel):
    id: UUID
    rider_name: Optional[str] = None
    rider_fee: Optional[float] = None
    deferral_bonus_rate: Optional[float] = None
    payout_percentage_single_age_65: Optional[float] = None
    payout_percentage_single_age_70: Optional[float] = None
    payout_percentage_single_age_75: Optional[float] = None
    payout_percentage_joint_age_65: Optional[float] = None
    payout_percentage_joint_age_70: Optional[float] = None
    payout_percentage_joint_age_75: Optional[float] = None
    lifetime_guarantee: Optional[bool] = None
    inflation_protection: Optional[bool] = None

    class Config:
        from_attributes = True


class IndexOptionResponse(BaseModel):
    id: UUID
    index_name: Optional[str] = None
    index_type: Optional[str] = None
    cap_rate: Optional[float] = None
    participation_rate: Optional[float] = None
    spread_fee: Optional[float] = None
    floor_rate: Optional[float] = None

    class Config:
        from_attributes = True


class HistoricalPerformanceResponse(BaseModel):
    id: UUID
    year: Optional[int] = None
    credited_rate: Optional[float] = None
    index_return: Optional[float] = None
    effective_return: Optional[float] = None

    class Config:
        from_attributes = True


class ProductResponse(BaseModel):
    id: UUID
    product_name: Optional[str] = None
    carrier_name: Optional[str] = None
    carrier_rating: Optional[str] = None
    product_type: Optional[str] = None
    state_availability: Optional[list] = None
    surrender_period: Optional[int] = None
    base_rate: Optional[float] = None
    cap_rate: Optional[float] = None
    participation_rate: Optional[float] = None
    annual_fee: Optional[float] = None
    has_bonus: Optional[bool] = None
    bonus_percentage: Optional[float] = None
    bonus_vesting_years: Optional[int] = None
    minimum_investment: Optional[int] = None
    maximum_investment: Optional[int] = None
    issue_ages_min: Optional[int] = None
    issue_ages_max: Optional[int] = None
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None

    class Config:
        from_attributes = True


class ProductListResponse(BaseModel):
    items: List[ProductResponse]
    total: int
    page: int
    limit: int
    total_pages: int
